pemutil/pem.go

```diff
@@ -231,52 +231,122 @@ func ParseCertificate(pemData []byte) (*x509.Certificate, error) {
 	return nil, errors.New("error parsing certificate: no certificate found")
 }
 
-// ParseCertificateBundle extracts all the certificates in the given data.
-func ParseCertificateBundle(pemData []byte) ([]*x509.Certificate, error) {
-	var block *pem.Block
-	var certs []*x509.Certificate
-	for len(pemData) > 0 {
-		block, pemData = pem.Decode(pemData)
-		if block == nil {
-			return nil, errors.New("error decoding pem block")
+// ParseCertificateBundle returns a list of *x509.Certificate parsed from
+// the given bytes.
+//
+// - supports PEM and DER certificate formats
+//   - DER format requires WithAllowDER() option
+//   - If a DER-formatted file is given only one certificate will be returned.
+func ParseCertificateBundle(data []byte, opts ...Options) ([]*x509.Certificate, error) {
+	var err error
+	// Populate options
+	ctx := newContext("")
+	if err := ctx.apply(opts); err != nil {
+		return nil, err
+	}
+
+	filename := "input"
+	if ctx.filename != "" {
+		filename = ctx.filename
+	}
+
+	// PEM format
+	if bytes.Contains(data, PEMBlockHeader) {
+		var block *pem.Block
+		var bundle []*x509.Certificate
+		for len(data) > 0 {
+			block, data = pem.Decode(data)
+			if block == nil {
+				break
+			}
+			if block.Type != "CERTIFICATE" {
+				continue
+			}
```

Comment:
[CertPool.AppendCertsFromPEM](https://cs.opensource.google/go/go/+/refs/tags/go1.22.5:src/crypto/x509/cert_pool.go;l=226-228) has the condition that you have removed:
```go
if block.Type != "CERTIFICATE" || len(block.Headers) != 0 {
	continue
}
```